docs/pages/pmd/userdocs/cli_reference.md

```diff
@@ -0,0 +1,265 @@
+
+<table>
+<tr><td>0</td><td>Everything is fine, no violations found</td></tr>
+<tr><td>1</td><td>Couldn't understand command-line parameters or PMD exited with an exception</td></tr>
+<tr><td>4</td><td>At least one violation has been detected, unless <code>-failOnViolation false</code> is set.</td></tr>
+</table>
+
+
+## Supported Languages
+
+*   [apex](pmd_rules_apex.html) (Salesforce Apex)
+*   [java](pmd_rules_java.html)
+*   [ecmascript](pmd_rules_javascript.html) (JavaScript)
+*   [jsp](pmd_rules_jsp.html)
+*   [plsql](pmd_rules_plsql.html)
+*   [vf](pmd_rules_vf.html) (Salesforce VisualForce)
+*   [vm](pmd_rules_vm.html) (Apache Velocity)
+*   [xml and xsl](/pmd_rules_xml.html)
+
+
+## Available Report Formats
+
+PMD comes with many different renderers.
+The mnemonics in bold are used to select them on the command line, as
+arguments to the `-format` option. Some formats accept *properties*,
+which can be specified with the `-property` option on the command-line.
+
+*   **codeclimate**: Renderer for Code Climate JSON format.
+
+*   **csv**: Comma-separated values tabular format.
+
+    Properties:
+
+    *   problem: Include problem column. Default: true.
+    *   package: Include package column. Default: true.
+    *   file: Include file column. Default: true.
+    *   priority: Include priority column. Default: true.
+    *   line: Include line column. Default: true.
+    *   desc: Include description column. Default: true.
+    *   ruleSet: Include Rule set column. Default: true.
+    *   rule: Include Rule column. Default: true.
+
+*   **emacs**: GNU Emacs integration.
+
+*   **html**: HTML format.
+
+    Properties:
+
+    *   linePrefix: Prefix for line number anchor in the source file.
+    *   linkPrefix: Path to HTML source.
+
+*   **ideaj**: IntelliJ IDEA integration.
+
+    Properties:
+
+    *   classAndMethodName: Class and method name, pass `.method` when processing a directory.
+    *   sourcePath:
+    *   fileName:
+
+*   **summaryhtml**: Summary HTML format.
+
+    Properties:
+
+    *   linePrefix: Prefix for line number anchor in the source file.
+    *   linkPrefix: Path to HTML source.
+
+*   **text**: Text format.
+
+*   **textcolor**: Text format, with color support (requires ANSI console support, e.g. xterm, rxvt, etc.).
+
+    Properties:
+
+    *   color: Enables colors with anything other than `false` or `0`. Default: yes.
+
+*   **textpad**: TextPad integration.
+
+*   **vbhtml**: Vladimir Bossicard HTML format.
+
+*   **xml**: XML format.
+
+    Properties:
+
+    *   encoding: XML encoding format, defaults to UTF-8.
+
+*   **xslt**: XML with a XSL transformation applied.
+
+    Properties:
+
+    *   encoding: XML encoding format, defaults to UTF-8.
+    *   xsltFilename: The XSLT file name.
+
+*   **yahtml**: Yet Another HTML format.
+
+    Properties:
+
+    *   outputDir: Output directory.
+
+
+## Incremental Analysis
```

Comment:
these Incremental Analysis sections feel kinda out of place under the CLI Reference page, as they don't refer exclusively to the CLI...

Also, remember the code references this section under the old structure, the link would have to be updated as part of the change to the website:

https://github.com/pmd/pmd/blob/15c2eb928b64cc92827c4b6be02a317bc4379d03/pmd-core/src/main/java/net/sourceforge/pmd/PMD.java#L310-L311

Reply:
You're right, but there's no obvious candidate for where to put this documentation. Maybe in the `installation.md`, though again, this is specific to CLI usage. We could also create a new short page for that.

Reply:
Maybe moving it out to a separate page on the topic?

Reply:
@jsotuyod I created a new page for that. I'm not too familiar with how maven or Ant support incremental analysis, but this could be added to that page.